cmd/traffic/cmd/manager/service.go

```diff
@@ -268,10 +272,98 @@ func (s *service) Remain(ctx context.Context, req *rpc.RemainRequest) (*empty.Em
 func (s *service) Depart(ctx context.Context, session *rpc.SessionInfo) (*empty.Empty, error) {
 	ctx = managerutil.WithSessionInfo(ctx, session)
 	sessionID := tunnel.SessionID(session.GetSessionId())
-
+	err := s.UpdateLastEngagementTime(ctx, sessionID)
 	// There's no reason for the caller to wait for this removal to complete.
 	go s.state.RemoveSession(context.WithoutCancel(ctx), sessionID)
-	return &empty.Empty{}, nil
+	return &empty.Empty{}, err
+}
+
+func (s *service) removeUnusedAgent(ctx context.Context, sessionID tunnel.SessionID) error {
+	maxIdleTime := managerutil.GetEnv(ctx).AgentMaxIdleTime
```

Comment:
I think this method should begin with `agentKey, err := s.getAgentKey(sessionID)` and simply return `nil` when there is no agent (i.e. the `sessionID` is a client session). No reason to go through the motion of unmarshaling the yaml first.